src/main/java/com/target/ready/library/system/Controller/LibraryController.java

```diff
@@ -18,10 +18,20 @@ public class LibraryController {
     LibraryController(LibrarySystemService librarySystemService){
         this.librarySystemService=librarySystemService;
     }
-    @GetMapping("getBooks")
-    public List<Book> getAllBooks(){
-
-        return librarySystemService.getAllBooks();
+    @GetMapping("books_directory")
+    public List<Book> getAllBooks(@RequestParam(value = "pageNumber", defaultValue = "0", required = false) Integer pageNumber){
```

Comment:
What will be the request parameter which will define the number of entries that needs to be returned in a page?

Reply:
No the number of entries is fixed by us in the backend . Request parameter is to get the pageNumber

Reply:
so if the user clicks on next button in the ui "current page number+1" is returned to backend and is fetched using Request param and corresponding entries is returned and similarly if user clicks on previous button "current page number-1" is returned to backend